tools/topology/topology1/sof-icl-rt711-rt1308-rt715-hdmi.m4

```diff
@@ -40,6 +40,46 @@ ifdef(`MIC_LINK',`',
 # HDMI ID calculated based on the configuraiton
 define(HDMI_BE_ID_BASE, `0')
 
+# define the codec endpoints order according to each codec dai index
+# assign rt711 dai index
+define(`ALH_JACK_OUT_INDEX', `0')
+define(`ALH_JACK_IN_INDEX', `1')
+define(`ALH_JACK_AMP_INDEX', `0')
+define(`ALH_DMIC_IN_INDEX', `0')
+
+define(`ALH_JACK_OUT_NAME', `SDW'eval(UAJ_LINK)`-Playback')
+define(`ALH_JACK_IN_NAME', `SDW'eval(UAJ_LINK)`-Capture')
+define(`ALH_AMP_OUT_NAME', `SDW'eval(AMP_1_LINK)`-Playback')
+define(`ALH_MIC_IN_NAME', `SDW'eval(MIC_LINK)`-Capture')
+
+ifdef(`MFC', `
+# assign cs42l43 dai index
+undefine(`ALH_JACK_OUT_INDEX')
+undefine(`ALH_JACK_IN_INDEX')
+undefine(`ALH_JACK_AMP_INDEX')
+undefine(`ALH_DMIC_IN_INDEX')
+define(`ALH_JACK_OUT_INDEX', `0')
+define(`ALH_JACK_IN_INDEX', `2')
+define(`ALH_JACK_AMP_INDEX', `3')
+define(`ALH_DMIC_IN_INDEX', `1')
+
+# align the partial string match of full dai link name
+undefine(`ALH_JACK_OUT_NAME')
+undefine(`ALH_JACK_IN_NAME')
+undefine(`ALH_AMP_OUT_NAME')
+undefine(`ALH_MIC_IN_NAME')
+define(`ALH_JACK_OUT_NAME', `Playback-SimpleJack')
+define(`ALH_JACK_IN_NAME', `Capture-SimpleJack')
+define(`ALH_AMP_OUT_NAME', `Playback-SmartAmp')
+define(`ALH_MIC_IN_NAME', `Capture-SmartMic')'
+)
+
+# SDW Pin as DAIs index starts from 2 on each SDW link
+define(`ALH_PIN2', `2' + `ALH_JACK_OUT_INDEX')
+define(`ALH_PIN3', `2' + `ALH_JACK_IN_INDEX')
+define(`ALH_PIN4', `2' + `ALH_JACK_AMP_INDEX')
+define(`ALH_PIN5', `2' + `ALH_DMIC_IN_INDEX')
```

Comment:
Why don't we just define ALH_JACK_OUT_PIN = 2 and use it?

Reply:
The PR moved to #9633 and merged on stable-v2.2 for Linux SKU. Close here.